codec-http2/src/main/java/io/netty/handler/codec/http2/HpackDecoder.java

```diff
@@ -436,6 +484,43 @@ private static boolean contains(Http2Headers headers, CharSequence name) {
         return false;
     }
 
+    private static boolean isValidHeaderValue(CharSequence value) {
+        // Validate value to field-content rule.
+        //  field-content  = field-vchar [ 1*( SP / HTAB ) field-vchar ]
+        //  field-vchar    = VCHAR / obs-text
+        //  VCHAR          = %x21-7E ; visible (printing) characters
+        //  obs-text       = %x80-FF
+        //  SP             = %x20
+        //  HTAB           = %x09 ; horizontal tab
+        //  See: https://datatracker.ietf.org/doc/html/rfc7230#section-3.2
+        //  And: https://datatracker.ietf.org/doc/html/rfc5234#appendix-B.1
+        int length = value.length();
+        if (length == 0) {
+            return true;
+        }
+        final byte[] array;
+        final int start;
+        if (value instanceof AsciiString) {
+            AsciiString ascii = (AsciiString) value;
+            array = ascii.array();
+            start = ascii.arrayOffset();
+        } else {
+            array = value.toString().getBytes(CharsetUtil.US_ASCII);
```

Comment:
hmm.. I wonder if it would be better to use `charAt(...)` in this case to reduce copy / allocation.

Reply:
I've added specialization for AsciiString and CharSequence, respectively, so we avoid allocating arrays for non-AsciiString inputs.